app/jobs/bsnl_sms_status_job.rb

```diff
@@ -0,0 +1,50 @@
+class BsnlSmsStatusJob
+  include Sidekiq::Worker
+  include Sidekiq::Throttled::Worker
+  BSNL_TIME_ZONE = "Asia/Kolkata"
+
+  sidekiq_options queue: :default
+  sidekiq_options retry: 3
+
+  sidekiq_throttle(
+    threshold: {limit: 250, period: 1.minute}
+  )
+
+  def perform(message_id)
+    response = Messaging::Bsnl::Api.new.get_message_status_report(message_id)
+    handle_api_errors(message_id, response)
+
+    detailable = BsnlDeliveryDetail.find_by!(message_id: message_id)
+    detailable.message_status = response["Message_Status"] if response["Message_Status"]
```

Comment:
We could do `response["Message_Status"].presence`

Reply:
We don't want to override a message status if it already exists, hence the `if`.

Reply:
Oh I see, very cool.